packages/mdx/lib/plugin/recma-jsx-rewrite.js

```diff
@@ -300,32 +301,24 @@ export function recmaJsxRewrite(options) {
             }
 
             if (defaults.length > 0 || parameters.length > 1) {
-              parameters.unshift({
-                type: 'ObjectExpression',
-                properties: defaults
-              })
+              for (const parameter of parameters) {
+                defaults.push({type: 'SpreadElement', argument: parameter})
+              }
             }
 
             // If we’re getting components from several sources, merge them.
             /** @type {Expression} */
             let componentsInit =
-              parameters.length > 1
-                ? {
-                    type: 'CallExpression',
-                    callee: toIdOrMemberExpression(['Object', 'assign']),
-                    arguments: parameters,
-                    optional: false
-                  }
-                : parameters[0].type === 'MemberExpression'
-                ? // If we’re only getting components from `props.components`,
+              defaults.length > 0
+                ? {type: 'ObjectExpression', properties: defaults}
+                : // If we’re only getting components from `props.components`,
                   // make sure it’s defined.
                   {
                     type: 'LogicalExpression',
                     operator: '||',
                     left: parameters[0],
                     right: {type: 'ObjectExpression', properties: []}
                   }
-                : parameters[0]
```

Comment:
This `else` was reached, but removing it doesn’t change the test output and I’m not sure what it’s for.

Reply:
I don’t remember and I don’t know from looking at just this PR